### DISCUSSION TITLE
How to run 2 workspaces with different poolOptions.threads.isolated option?

### DISCUSSION THREAD

**Discussion by ArtemZag** [vitest-dev/vitest#4703] [Q&A] [answered]
poolOptions.threads.isolated=false is very unstable but provides significantly better performance.

So, I'd like to create a new workspace with different poolOptions defined for it.
But looks like poolOptions can only be defined globally and I can only select 'thread' | 'fork' or 'vmThread' per workspace.

Is there any workaround to run one workspace in isolated mode and another one in non-isolated ?

**Comment by sheremet-va** [comment]
You cannot, `poolOptions` can only be configured at the top-level config.

**Reply by ArtemZag** [reply]
And no plans to make it possible like it was before 1.x version?

**Reply by sheremet-va** [reply]
It was not possible to do before 1.x version, `isolated` in workspaces was ignored

**Reply by sheremet-va** [reply]
Actually, try setting `poolOptions.threads.singleThread` to `true`

**Reply by ArtemZag** [reply]
@sheremet-va , I did it on 0.34.1, then upgraded to 1.0.2 and noticed that poolOptions are completely ignored in workspaces.
Even if I reference different "vitest.config" file via "extend" property only poolOptions from main "vitest.config" file is considered.

I mean something like this:

```
defineWorkspace([
  {
    extends: "./vitest.config.noisolate.ts", -- has different poolOptions, but it's ignored
    ...
  },
  {
    extends: "./vitest.config.ts"
    ...
  }
])
```

**Reply by sheremet-va** [reply]
Yeah, I see. `poolOptions` generally can not be inherited. Only `singleThread`/`singleFork` can in theory have some effect, but it's also not inherited. @AriPerkkio what do you think?

**Reply by AriPerkkio** [reply]
Looks like we are already supporting `environment` and `environmentOptions` in workspace project level. That means that we should be able to support `pool` and `poolOptions` too, right?

**Reply by sheremet-va** [reply, marked as answer]
Yes, they were previously supported. (At least some of the options). I don't think we can support most options, but we can definitely support `singleFork`/`singleThreads` and different pools. Actually pools implementation already assumes it can be configured differently, we just need to pass it down correctly.

**Reply by ArtemZag** [reply]
@sheremet-va , should I create an issue based on this discussion?

**Reply by sheremet-va** [reply]
Yes
